faq.html

```diff
@@ -161,6 +161,87 @@ <h2 id="outputformat">The output format of ParticleFile</h2>
+  <ul>
+    <li>Initialize your ParcileSet with only few particles, e.g. <code class="language-python">pset = ParticleSet(fieldset=fieldset, pclass=JITParticle, lon=np.random.rand(16,1), lat=np.random.rand(16,1))</code></li>
+    <li>Do not re-add particles periodically via <code class="language-python">repeatdt</code>, e.g. change <code class="language-python">pset = ParticleSet(..., lat=np.random.rand(16,1), repeatdt=timedelta(hours=1))</code> to <code class="language-python">pset = ParticleSet(..., lat=np.random.rand(16,1))</code></li>
+  </ul>
+  Run your simulation again and observe the memory with the tools mentioned above. If your simulation runs through without problems, it is almost certain your real simulation requires too many particles. There is no generic solution at this point for it, so: Get in touch with us at <a href="https://github.com/OceanParcels/parcels/issues">the OceanParcels GitHub page</a> via the issue tracker to find an individual solution.
+  <p></p>
+  If the reduction of particles indeed does not solve your issue and your simulation still crashes, then this is likely due to the Fields consuming more memory than available. Luckily, there are some solutions to this problem.
+  <p></p>
+
+  <h4>Simulations crash on HPC clusters (via job submission systems such as SGE or SLURM), but locally finishes without problems</h4>
+  <p></p>
+  The behaviour of simulations crashing on cluster systems whereas running as-expected on local machines can occur because
+  <ul>
+    <li>Python itself does not return memory to the operation system</li>
+    <li>your (field) data are too large for the memory, but fit comfortably in <i>virtual memory</i> (e.g. swap drive, swap files, page files) =-> job submission systems do not provide access to the <i>swap</i> memory</li>
+    <li>your (field) data exceed the default memory capacity granted to users of your cluster -> consult tutorials on <b>SGE qsub</b> (e.g. from <a href="http://star.mit.edu/cluster/docs/0.92rc2/guides/sge.html">MIT</a> and <a href="https://wikis.utexas.edu/display/CCBB/sge-tutorial">University of Austin / Texas</a>) or <b>SLURM</b> (e.g. from <a href="https://researchcomputing.princeton.edu/education/online-tutorials/getting-started/introducing-slurm">Princeton University</a> or <a href="https://ubccr.freshdesk.com/support/solutions/articles/5000688140-submitting-a-slurm-job-script">University of Buffalo</a>) to raise the granted memory to your simulation job. In particular, consider submission option parameters such as <i>-l h_vmem=<b>size</b></i> (SGE qsub) and <i>
+      --mem=<b>size</b></i> (SLURM).</li>
+  </ul>
+
+  <p></p>
+  <h4>Introduce field chunking (i.e. <i>dynamic loading</i> of fields)</h4>
```

Comment:
Why not put this at the top? I understand that you want users to go through the forensics of their issue first, but we can expect field chunking to solve a lot of problems. So I would say we provide that as the first option?

Reply:
Yes, good argument, I agree.